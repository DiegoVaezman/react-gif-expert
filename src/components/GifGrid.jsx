import React, { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropsTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    const { gifs, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            <div className='card-grid'>
                {isLoading && <p>Cargando...</p>}
                {gifs.map((gif) => (
                    <GifItem key={gif.id} {...gif} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropsTypes.string.isRequired,
};
