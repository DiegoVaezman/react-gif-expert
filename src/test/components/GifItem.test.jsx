import { render, screen } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('Pruebas en <GifItem />', () => {
    const title = 'test title';
    const url = 'testurl';

    test('should render props title and url ', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        expect(screen.getByText(title)).toBeTruthy();

        const { src, alt } = screen.getByRole('img');
        expect(src).toContain(url);
        expect(alt).toBe(title);
    });

    test('should match to snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });
});
