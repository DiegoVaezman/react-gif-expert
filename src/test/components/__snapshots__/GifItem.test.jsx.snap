// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> should match to snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test title"
      data-testid="test-img"
      src="testurl"
    />
    <p>
      test title
    </p>
  </div>
</div>
`;
